File: 03-poo/09-interfaces-y-traits/scala/soluciones/RetoExportable.scala
trait ExportableReto:
  def exportar: String
class Csv extends ExportableReto:
  def exportar: String = "datos,csv"
class Json extends ExportableReto:
  def exportar: String = "{\"datos\":true}"

@main def retoExportable(): Unit = List[ExportableReto](new Csv,new Json).foreach(r => println(r.exportar))

File: 03-poo/09-interfaces-y-traits/java/soluciones/RetoExportable.java
public class RetoExportable {
    interface Exportable { String exportar(); }
    static class ReporteCsv implements Exportable { public String exportar(){ return "datos,csv"; } }
    static class ReporteJson implements Exportable { public String exportar(){ return "{\"datos\":true}"; } }
    public static void main(String[] args){ Exportable[] reportes={new ReporteCsv(),new ReporteJson()}; for(Exportable r:reportes) System.out.println(r.exportar()); }
}
